packages/react-server/core/FramebackController.js

```diff
@@ -289,12 +289,15 @@ class FramebackController extends EventEmitter {
 		// If the frame has a client controller we'll listen to when
 		// _it_ tells us it's done loading.
 		//
+		if (clientController && !clientController.__parentIsListening) {
+			clientController.__parentIsListening = true;
+			clientController.context.onLoadComplete(this._handleFrameLoad.bind(this, frame));
+		}
+
 		// It may have finished _before_ we get here, so we'll also check
 		// whether it has already set its `_previouslyRendered` flag.
 		//
-		if (clientController && !clientController.__parentIsListening && !clientController._previouslyRendered) {
-			clientController.__parentIsListening = true;
-			clientController.context.onLoadComplete(this._handleFrameLoad.bind(this, frame));
+		if (clientController && !clientController._previouslyRendered) {
```

Comment:
Basically: We _always_ need to wire up our listener (_once_).  And, in the case where the event we're listening to hasn't fired yet, we'll bail out early and wait for it.